packages/next/src/client/components/react-dev-overlay/pages/hooks.ts

```diff
@@ -0,0 +1,53 @@
+import type { ErrorType } from '../types'
+import { useCallback, useEffect } from 'react'
+import { useErrorOverlayReducer } from '../shared'
+import { on, off } from './bus'
+
+const shouldPreventDisplay = (
+  errorType?: ErrorType | null,
+  preventType?: ErrorType[] | null
+) => {
+  if (!preventType || !errorType) {
+    return false
+  }
+  return preventType.includes(errorType)
+}
+
+export const usePagesReactDevOverlay = (preventDisplay?: ErrorType[]) => {
```

Comment:
Ported from `packages/next/src/client/components/react-dev-overlay/pages/ReactDevOverlay.tsx` as a hook to be used easily on v2